File: CodeForces/tetris.kt
fun main(arg: Array<String>){
	val (n, m) = readLine().toString().split(" ").map{it.toInt()}
	val squares = readLine().toString().split(" ").map{it.toInt()}
	val finish = Array(n){ 0 }
	squares.forEach(){
		finish[it-1]++
	}
	println(finish.min())
}

File: CodeForces/tetris.py
'''
3 9
1 1 2 2 2 3 1 2 3
'''
from collections import Counter
cols, squares = [int(x) for x in input().split()]
colsarr = [0]*cols
squarr  = Counter([int(x)-1 for x in input().split()])
if len(squarr) == cols:
    print(min(squarr.values()))
else:
    print(0)

